src/main/kotlin/cacophony/codegen/instructions/CacophonyInstructionCovering.kt

```diff
@@ -2,14 +2,41 @@ package cacophony.codegen.instructions
 
 import cacophony.codegen.BlockLabel
 import cacophony.codegen.instructions.matching.InstructionMatcher
+import cacophony.codegen.instructions.matching.Match
 import cacophony.controlflow.CFGNode
+import cacophony.controlflow.Register
+
+class CacophonyInstructionCovering(private val instructionMatcher: InstructionMatcher) : InstructionCovering {
+    private fun coverGivenMatch(match: Match): List<Instruction> {
+        val tempRegisters = match.toFill.keys.associateWith { Register.VirtualRegister() }
```

Comment:
NIT: I think you could replace `keys.associateWith` with `mapValues` (to make it more idiomatic)